cli/import_map.rs

```diff
@@ -197,12 +197,28 @@ impl ImportMap {
     // the url since it contains what looks to be a uri scheme. To work around
     // this, we append the specifier to the path segments of the base url when
     // the specifier is not relative or absolute.
+    let mut maybe_query_string = None;
     if !is_relative_or_absolute_specifier && base.path_segments_mut().is_ok() {
       {
         let mut segments = base.path_segments_mut().unwrap();
         segments.pop_if_empty();
-        segments.extend(specifier.split('/'));
+
+        // Handle query string first, otherwise it would be percent-encoded
+        // by `extend()`
+        let mut parts: Vec<&str> = specifier.split('?').collect();
+        assert!(parts.len() == 1 || parts.len() == 2);
```

Comment:
With this change, `import "key/a?b?c"` would panic.

Reply:
Is it actually valid specifier? Shouldn't it contain at most single `?` and the rest should be percent encoded?

Reply:
It's perfectly valid for a URL query string to contain `?`:

```
> new URL("https://example.com/a?b?c")
URL {
  href: "https://example.com/a?b?c",
  origin: "https://example.com",
  protocol: "https:",
  username: "",
  password: "",
  host: "example.com",
  hostname: "example.com",
  port: "",
  pathname: "/a",
  hash: "",
  search: "?b?c"
}
```

And even with a character like `<`, which does get percent-encoded in the query string, you'd reach step 1.2.5 in the spec's "resolve an imports match" algorithm, which would parse the specifier minus the prefix. The URL parsing wouldn't fail – rather than that, the serialization of _url_ then would have that character percent-encoded.

Reply:
Doesn't the spec mandate how to handle this case?

Reply:
Per the spec, the `append_specifier_to_base` function should just do `base.join(specifier)`. The complication comes because we accept specifiers which are Windows paths. (I'm not sure why we must support them as suffixes of an import map match, but whatever.)

In any case, since Windows filenames don't support the question mark, so assuming we don't support UNC paths (which start with `\\?\`), Windows paths won't contain it either. So we can just split at the first question mark.

Also, what if the specifier contains a fragment?

Reply:
> So we can just split at the first question mark.

So you're suggesting to just treat everything after the first question mark as a "query string"?

> Also, what if the specifier contains a fragment?

Good point, we should probably also split on first hash (`#`) and append it manually.

It's unfortunate that handling Windows paths adds so much complexity and as can be seen in this example bugs.

Reply:
> So you're suggesting to just treat everything after the first question mark as a "query string"?

How about splitting the specifier by the first either question mark or hash, and if there's a suffix, parse it with the extended base:

```rust
let prefix = match specifier.find(&['?', '#']) {
  Some(idx) => {
    maybe_query_string = Some(&specifier[..idx]);
    &specifier[..idx]
  },
  None => specifier
};
segments.extend(prefix.split('/'));

// ...

if let Some(query_string_and_fragment) = maybe_query_string {
  Ok(base.join(query_string_and_fragment))
} else {
  Ok(base)
}
```

Reply:
Thanks Andreu, I applied you suggestions and added you as co-author